src/FissionErrors.ts

```diff
@@ -0,0 +1,73 @@
+export type ByteLength = number
+
+export interface AxiosError extends Error {
+    isAxiosError: boolean
+    config: {
+        headers: {
+            "Content-Length": ByteLength
+        }
+        maxContentLength: ByteLength
+    }
+}
+
+export class FileSizeError extends Error {
+  fileSizeBytes: ByteLength
+  maxFileSizeBytes: ByteLength
+  details: string
+  private axiosError: AxiosError
+  private AXIOS_DEFAULT_MAXIMUM = 10_000_000;
+
+  // Impossible to cover branch on super()
+  /* istanbul ignore next */
+  constructor(axiosError: AxiosError) {
+      super("File size limit exceeded");    
+      
+      // https://github.com/facebook/jest/issues/8279
+      Object.setPrototypeOf(this, Error.prototype);
+      
+      this.axiosError = axiosError;
+      this.fileSizeBytes = this.extractFileSize()
+      this.maxFileSizeBytes = this.extractMaxFileSize();
+      this.details = this.constructDetails();
+    }
+  
+  private extractFileSize = () => {
+    return this.axiosError.config.headers["Content-Length"];
+  }
+
+  private extractMaxFileSize = (): ByteLength => {
+    let maxFileSizeBytes = this.axiosError.config.maxContentLength;
+    
+    if(maxFileSizeBytes === -1) {
+      maxFileSizeBytes = this.AXIOS_DEFAULT_MAXIMUM;
+    }
+    
+    return maxFileSizeBytes;
+  }
+  
+  private humanReadable = (bytes: number): string => {
```

Comment:
We should limit class methods as much as possible.

We should:
1. Move the division to a named utility function in the `utils.ts` file
2. leave the `MB` in the error message

```js
// util.js
const byteToMBSize = (bytes) => (bytes / 1_000_000).toFixed(1)
```

```js
    
    return `---
Oh no, the file you tried to add is too big 😲
You tried sending ${fileSize} MB but your current max is ${maxSize}.
That means your file was ${overage} MB too big 😟
To solve this, you may want to try:
 * Compressing your file
 * Breaking the file into smaller files
 * Contacting the nice people at Fission for assistance
Warm Regards and sorry for the inconvenience,
The Fission Devs 🤗
---

Reply:
Since `toFixed()` returns a string, I'm tempted to do something like this:
```js
const bytesToMegabytesConversion = (bytes: number): number => (bytes / 1_000_000)
const magebytesToGigabytes = (megabytes: number): number => (megabytes / 1_000)

const humanReadableDataSize = (bytes: number, decimals: number): string => {
    let units = "MB"
    let value: number = bytesToMegabytesConversion(bytes)
    if(value > 1_000) {
        units = "GB"
        value = magebytesToGigabytes(value)
    }
    return value.toFixed(decimals) + units
}
```